packages/rich-list-core/src/lib/Queue.ts

```diff
@@ -0,0 +1,45 @@
+export type Mode = 'LIFO' | 'FIFO'
+
+export interface Queue<T> {
+  push: (message: T) => Promise<string> // message id / handle
+  receive: (maxMessage: number) => Promise<T[]>
+}
+
+export interface QueueClient<T> {
+  createQueueIfNotExist: (name: string, mode: Mode) => Promise<Queue<T>>
+}
+
+/**
+ * For unit test use.
+ */
+export class InMemoryQueueService<T> implements Queue<T> {
+  private DATA: T[] = []
+
+  async push (message: T): Promise<string> {
+    this.DATA.push(message)
+    return `${this.DATA.length - 1}`
+  }
```

Comment:
```suggestion
  async push (message: T): number {
    this.DATA.push(message)
    return this.DATA.length - 1
  }
```

Reply:
Yup, it should be a promise, grabbing from an array is solely this test implementation.
And the returned handle to be string is more generic and conventional queue service like SQS will have that.

Reply:
ohh.. missed that.. okay.. because it was used for in-memory ops.. so confuse the `async` here